src/modules/hip/kernel/resample.hpp

```diff
@@ -0,0 +1,178 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 - resample kernel --------------------
+// Single channel resample support
+__global__ void resample_1channel_hip_tensor(float *srcPtr,
+                                             float *dstPtr,
+                                             float inRateTensor,
+                                             float outRateTensor,
+                                             int srcLength,
+                                             int outEnd,
+                                             double scale,
+                                             RpptResamplingWindow &window)
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+
+    int block = 256;
+    int outBlock = id_x * block;
+    if (outBlock >= outEnd)
+        return;
+
+    int blockEnd = std::min(outBlock + block, outEnd);
+    double inBlockRaw = outBlock * scale;
+    int inBlockRounded = static_cast<int>(inBlockRaw);
+    float inPos = inBlockRaw - inBlockRounded;
+    float fscale = scale;
+
+    for (int outPos = outBlock; outPos < blockEnd; outPos++, inPos += fscale)
+    {
+        int loc0, loc1;
+        window.input_range(inPos, &loc0, &loc1);
```

Comment:
I think in DALI's implemenation shared memory is used for window 
https://github.com/NVIDIA/DALI/blob/main/dali/kernels/signal/resampling_gpu.cuh#L76

Since window lookup is accessed by multiple threads and there is a chance of repetitive access to same memory in lookup, shared memory might be beneficial here. Please check and see if we call also incorporate something similar in our code